c++/src/kj/refcount.h

```diff
@@ -145,6 +148,14 @@ Rc<T> Refcounted::addRcRefInternal(T* object) {
   return Rc<T>(object);
 }
 
+template<typename T>
+struct RcDisposer {
+  inline static void dispose(void* value) {
```

Comment:
Static disposers don't need to deal with `void*`.

You can make this:

```suggestion
  inline static void dispose(kj::Refcounted* value) {
```

And then remove the `reinterpret_cast`, and also remove the template argument from `RcDisposer`.

This also lets you use `kj::mv()` when upcasting, so you don't have to deal with `disown`.